sst/test/include/framework/process.hh

```diff
@@ -0,0 +1,254 @@
+#pragma once
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+#include "framework/output.hh"
+
+/**
+ * This struct holds all information related to any
+ * errors/exceptions that could happen during execution of the binary specified
+ * by a TEST_GROUP. A process exception is thrown upon failure and handled
+ * inside the exception handler.
+ */
+struct ProcessException {
+  /** Custon error string specifying the type of error. */
+  std::string errString_;
+  /** Captured stdout of the child process. */
+  std::string stderrStr_;
+  /** Captured stderr of the child process. */
+  std::string stdoutStr_;
+  ProcessException() : errString_(""), stderrStr_(""), stdoutStr_(""){};
+  ProcessException(std::string errString)
+      : errString_(errString), stderrStr_(""), stdoutStr_(""){};
+  ProcessException(std::string errString, std::string stderrStr,
+                   std::string stdoutStr)
+      : errString_(errString), stderrStr_(stderrStr), stdoutStr_(stdoutStr){};
+};
+
+/**
+ * Process class responsible for creating a child process which executes
+ * SST with the configuration file and CLI arguments specified by TEST_GROUP(s)
+ * and TEST_CASE(s). This class also captures the stdout and stderr of the child
+ * process and passes it back to the main process.
+ */
+class Process {
+ private:
+  /** Captured stdout of the child process. */
+  std::string stdoutCapture_;
+  /** Captured stderr of the child process.*/
+  std::string stderrCapture_;
+  /** Reference to the output class.*/
+  Output output_;
+  /** Default CLI argument specified by a TEST_GROUP. */
+  std::vector<std::string> defaultCliArgs_;
+  /** Path to the SST binary. */
+  std::string sstBinPath_;
+  /** Path to the SST config file defined by the TEST_GROUP. */
+  std::string sstSimConfigFile_;
+  /** Command used to invoke the SST binary. */
+  std::string sstCmd_;
+
+ public:
+  Process(std::vector<std::string> args) {
+#ifdef SST_TEST_CMD
+    sstCmd_ = SST_TEST_CMD;
+#else
+    sstCmd_ = "sst";
+#endif
+    sstSimConfigFile_ = std::string(SST_CONFIG_DIR) + "/" + args.at(1);
+    sstBinPath_ = std::string(SST_INSTALL_DIR) + "/bin/" + sstCmd_;
+    args.erase(args.begin());
+    args.erase(args.begin());
+    defaultCliArgs_ = args;
+  };
+  Process() {}
+
+  /**
+   * This method is used to run the SST executable with a config file and cli
+   * arguments. This method takes in additional cli arguments which replace
+   * default ones. If no additional cli arguments are given the default ones are
+   * used.
```

Comment:
Probably better to not use `cli` in comments

Reply:
Changed.